### DISCUSSION TITLE
Custom validation and form designer

### DISCUSSION THREAD

**Discussion by justynpride** [HiEventsDev/Hi.Events#135] [Ideas]
I was thinking about tickets, questions and custom validations per ticket. I like what is in place, but know there are some helpful additions features that would make it really powerful. This largely is around custom validations per ticket and form layout. 

For example a question requiring age might have different validations on different tickets - ie age must be within different date ranges for different kids tickets. 

Another example would be where in response to an option being selected or a box being checked, a field might then need to be displayed and be required.

An event may want a different ordering of the questions on the tickets due to different logic.

I’ve seen different systems where this is done within the ticket (like a form designer), or another where the form designer is separate and you then link the form to the ticket. The latter option feels more cumbersome.

Putting the idea out for discussion and roadmap consideration.

**Comment by daveearley** [comment]
Good suggestions! There is definitely a lot of improvements that could be added around questions. Eventbrite has the concept of conditional "sub questions", which I think would work well.

<img width="984" alt="Screenshot 2024-08-10 at 19 38 05" src="https://github.com/user-attachments/assets/b696789b-08f3-42a2-a661-1ae2d3cce964">

The custom validations are a nice idea too. The UI would have to be well though out so as not to add too much complexity.

**Reply by justynpride** [reply]
I agree it is going to be vital to keep and UI clear and concise. 

It would be interesting to hear how 'complex question' requirements are for other people. I've done events where there are people registering to work with kids or other teams , and so the complexity has come more from that. There are groups of questions that only relate to that team, and then other questions that only relate to the other teams.

Is an option to have the questions module as it is where you create all the questions and allocation to tickets as it currently is. Then have an advanced 'ticket layout editor', where for each ticket you can drag to reorder the questions, and allocate as sub questions with the relevant validation questions? 

I don't know if this adds complexity, or allows the ability for 'simple ticketing', and 'advanced ticketing'?
